rust/src/lib/nm/settings/ip.rs

```diff
@@ -86,6 +99,19 @@ fn gen_nm_ipv4_setting(
     if let Some(rules) = iface_ip.rules.as_ref() {
         nm_setting.route_rules = gen_nm_ip_rules(rules, false)?;
     }
+    if let Some(setting) = nm_conn.connection.as_ref() {
+        if let Some(iface_name) = setting.iface_name.clone() {
+            if iface_name.eq(&"lo") {
+                if let Some(rules) = iface_ip.rules.as_ref() {
+                    nm_setting.replace_local_rule =
+                        Some(!rules_contains_local_rule(rules, false));
```

Comment:
This is wrong. Changes to local route rule should not set `replace_local_rule`. We should only set `nm_setting.replace_local_rule` when user desire state has

```
priority: 0
state: absent
```

Other cases, we are expecting user to append route rule and do not touch current value of `nm_setting.replace_local_rule`.

Reply:
You are mixing the `local route` and `priority 0`. This PR is focusing on `priority 0`.

* To delete priority 0 route rule, user can do `priority: 0` and `state: absent`. No matter user add another local route or not, we set `NM.replace_local_rule`  to true.

* To add it back, user can just do `prioirity: 0` with whatever detail they want, if the desire appen to be matching with `priority 0, route: 255`, we use kernel autogenerate one by set `NM.replace_local_rule` to false, if not, we still do `replace_local_rule: true` and manual rule.

Reply:
No I do not think so. `local route rule` and `priority 0` is not a problem. We already support them as expected.

There is an special `rule` created by kernel which is place on the local route table (255) with priority 0. This specific rule requires special treatment. IMHO we should drop it only when the user explicitly specify it e.g

```
priority: 0
route-table: 255
state: absent
family: ipv4
```

also for family ipv6. Please, notice that we shouldn't do the usual matching we do here. If an user specify only `priority: 0` or `route-table: 255` as absent we shouldn't drop this special rule. In most of the cases without this rule the user will lose their connectivity.